Use a single memory map for Blackhole ETH

nhuang-tt opened the pull request:
### Ticket
Link to Github Issue

### Problem description
Maintain single source of truth for the memory map

### What's changed
- Put all active/idle erisc mem map values in dev_mem_map
- eth_l1_address map is kept for backward compatibility

### Checklist
BH
https://github.com/tenstorrent/tt-metal/actions/runs/14805495792
https://github.com/tenstorrent/tt-metal/actions/runs/14867753889

abhullar-tt commented on tt_metal/hw/firmware/src/slave_active_erisc.cc:
```diff
@@ -0,0 +1,108 @@
+// clang-format on
+
+tt_l1_ptr mailboxes_t* const mailboxes = (tt_l1_ptr mailboxes_t*)(MEM_AERISC_MAILBOX_BASE);
+volatile tt_l1_ptr uint8_t* const slave_idle_erisc_run = &mailboxes->slave_sync.dm1;
```
Wondering if we need this file at all or if we can just use commonize idle_erisc fw for both

nhuang-tt replied:
I removed this. I think we can just use 1.

abhullar-tt commented on tt_metal/hw/firmware/src/slave_active_erisc.cc:
```diff
@@ -0,0 +1,108 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
```
If this pr is going in separately from full support could you add this in a separate pr and just have the mem map changes here

nhuang-tt replied:
Ack

nhuang-tt committed:
Single source of truth for BH's ethernet memory map

- Consolidate entries into dev_mem_map
- eth_l1_address_map left for backward compaitibility

nhuang-tt commented on tt_metal/hw/inc/blackhole/dev_mem_map.h:
```diff
@@ -135,31 +135,66 @@
+#define MEM_ERISC_BARRIER_BASE (MEM_ERISC_APP_ROUTING_INFO_BASE + MEM_ERISC_APP_ROUTING_INFO_SIZE)
+
+// Common Misc
+#define MEM_RETRAIN_COUNT_ADDR 0x1EDC
```
New

abhullar-tt replied:
Rather than listing out individual addresses I think it might be better to store the base address for the entire boot_results_t struct

nhuang-tt replied:
Boot results is at `MEM_SYSENG_ETH_RESULTS_BASE_ADDR`. I removed api table and mailbox base as those can be accessed through boot results.

The MSG ones are for sending messages

abhullar-tt replied:
wdyt about just keeping `MEM_SYSENG_ETH_RESULTS_BASE_ADDR` and adding the newly added eth_api_table_t structs and eth_mailbox_e enum in something like `umd/device/types/blackhole_eth.h` that we can then use across host + device

This could reduce manually needing to update addresses but we would need to keep this in sync with any FW changes anyways

nhuang-tt replied:
Sounds good. I didn't know we had this in umd. If the structs were in a syseng repo we could use them as a dependency but I guess we need to copy-paste for now.

abhullar-tt replied:
yeah, it would be best to bring in as a dependency (see https://tenstorrent.atlassian.net/wiki/spaces/syseng/pages/904626206/ETH+SW+APIs)

nhuang-tt replied:
I added `eth_api_table_t should only be included on the device to call functions in the base FW. It has pointers which could be the different on host vs. device.

nhuang-tt replied:
In my child branch `nhuang/bh-eth-dm2` I made the messages into a enum. The messages and mailbox address can be queried from the Host using the HAL. It returns 0 if not set (eg. wormhole).

E..g,
```
HalL1MemAddrType::ETH_FW_MAILBOX_MSG
HalL1MemAddrType::ETH_FW_MAILBOX_ARG0, ARG1, ARG2

...

     fw_mailbox_addr[utils::underlying_type<FWMailboxMsg>(FWMailboxMsg::ETH_MSG_CALL)] = MEM_SYSENG_ETH_MSG_CALL;
     fw_mailbox_addr[utils::underlying_type<FWMailboxMsg>(FWMailboxMsg::ETH_MSG_DONE)] = MEM_SYSENG_ETH_MSG_DONE;
     fw_mailbox_addr[utils::underlying_type<FWMailboxMsg>(FWMailboxMsg::ETH_MSG_LINK_STATUS_CHECK)] =
         MEM_SYSENG_ETH_MSG_LINK_STATUS_CHECK;
     fw_mailbox_addr[utils::underlying_type<FWMailboxMsg>(FWMailboxMsg::ETH_MSG_RELEASE_CORE)] =
         MEM_SYSENG_ETH_MSG_RELEASE_CORE;
```

There's also a function for writing to the mailbox address which is a no-op if the address is 0.

abhullar-tt replied:
nice this looks clean

nhuang-tt committed:
Add k_SingleProcessorMailboxSize to calculate mailbox size

nhuang-tt committed:
Add eth_fw_api.h for BH Base FW structs

abhullar-tt reviewed (approved):
lgtm, maybe in the PR to commonize active_erisc.cc and idle_erisc.cc we can see whether the idle/active mem addrs can be commonized as well
